chain/indexer/manager.go

```diff
@@ -0,0 +1,204 @@
+		api:      api,
+		storage:  strg,
+		indexer:  indexer,
+		exporter: NewModelExporter(),
+		window:   0,
+		pool:     workerpool.New(1),
+	}
+	for _, opt := range opts {
+		opt(im)
+	}
+	return im, nil
+}
+
+// TipSetAsync enqueues `ts` into the Manager's worker pool for processing. An error is returned if any Indexer in
+// the pool encounters a fatal error, else the method returns immediately.
+func (i *Manager) TipSetAsync(ctx context.Context, ts *types.TipSet) error {
+	if i.fatal != nil {
+		defer i.pool.Stop()
+		return i.fatal
+	}
+
+	stats.Record(ctx, metrics.IndexManagerActiveWorkers.M(i.active))
+	stats.Record(ctx, metrics.IndexManagerWaitingWorkers.M(int64(i.pool.WaitingQueueSize())))
+	if i.pool.WaitingQueueSize() > i.pool.Size() {
+		log.Warnw("queuing worker in Manager pool", "waiting", i.pool.WaitingQueueSize())
+	}
+	log.Infow("submitting tipset for async indexing", "height", ts.Height(), "active", i.active)
+
+	ctx, span := otel.Tracer("").Start(ctx, "Manager.TipSetAsync")
+	i.pool.Submit(func() {
+		atomic.AddInt64(&i.active, 1)
+		defer func() {
+			atomic.AddInt64(&i.active, -1)
+			span.End()
+		}()
+
+		ts := ts
+		success, err := i.TipSet(ctx, ts)
+		if err != nil {
+			log.Errorw("index manager suffered fatal error", "error", err, "height", ts.Height(), "tipset", ts.Key().String())
+			i.fatal = err
+			return
+		}
+		if !success {
+			log.Warnw("index manager failed to fully index tipset", "height", ts.Height(), "tipset", ts.Key().String())
+		}
+	})
+	return nil
+}
+
+// TipSet synchronously indexes and persists `ts`. TipSet returns an error if the Manager's Indexer encounters a
+// fatal error. TipSet returns false if one or more of the Indexer's tasks complete with a status `ERROR` or `SKIPPED`, else returns true.
+// Upon cancellation of `ctx` TipSet will persist all incomplete tasks with status `SKIPPED` before returning.
+func (i *Manager) TipSet(ctx context.Context, ts *types.TipSet) (bool, error) {
+	ctx, span := otel.Tracer("").Start(ctx, "Manager.TipSet")
+	defer span.End()
+	log.Infow("index tipset", "height", ts.Height())
+
+	var cancel func()
+	var procCtx context.Context // cancellable context for the task
+	if i.window > 0 {
+		// Do as much indexing as possible in the specified time window (usually one epoch when following head of chain)
+		// Anything not completed in that time will be marked as incomplete
+		procCtx, cancel = context.WithTimeout(ctx, i.window)
+	} else {
+		// Ensure all goroutines are stopped when we exit
+		procCtx, cancel = context.WithCancel(ctx)
+	}
+	defer cancel()
+
+	// asynchronously begin indexing tipset `ts`, returning results as they become avaiable.
+	taskResults, taskErrors, err := i.indexer.TipSet(procCtx, ts)
+	// indexer suffered fatal error, abort.
+	if err != nil {
+		return false, err
+	}
+
+	// there are no results, bail.
+	if taskResults == nil {
+		return true, nil
+	}
+
+	success := true
+	ctx, exportSpan := otel.Tracer("").Start(ctx, "Manager.Export")
```

Comment:
```suggestion
	ctx, exportSpan := otel.Tracer("").Start(ctx, "Manager.TipSet")
```

Reply:
actually just going to remove this, was residual from previous impl and the method already has a span.